### PR TITLE
Only osx - libprotobuf4233_libgrpc156

### PR DESCRIPTION
**Pull request by hmaarrfk**
Trying to debug a failure that historically occurs at:
```
  [3478/4743] cd /Users/runner/miniforge3/conda-bld/pytorch-recipe_1692633544754/work/build/third_party/onnx && /Users/runner/miniforge3/conda-bld/pytorch-recipe_1692633544754/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/python /Users/runner/miniforge3/conda-bld/pytorch-recipe_1692633544754/work/third_party/onnx/onnx/gen_proto.py -p onnx_torch -o /Users/runner/miniforge3/conda-bld/pytorch-recipe_1692633544754/work/build/third_party/onnx/onnx onnx -m
```

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
I need a local x86 mac to troubleshoot this better.

Essentially, we need to tell protoc to use the `BUILD_PREFIX` protoc during cross compilation, not the HOST one.

maybe i can troubleshoot using arm + linux

**Comment by h-vetinari** [comment]
That should be a common problem with a common solution though? We set protobuf to the build prefix for osx-arm all over the place. There's also a dedicated cmake variable for that.

File: recipe/build_pytorch.sh [hunk 1]
```diff
@@ -69,6 +76,19 @@ export USE_SYSTEM_SLEEF=1
...
+    rm -rf ${PREFIX}/bin/protoc-*
+
+    cp ${BUILD_PREFIX}/bin/protoc ${PREFIX}/bin/protoc
+    # libprotobuf 4.23.3 ships out the file protoc-23.3.0
+    # for example and refers to it explicitely
+    # not very portable, but i want to get through the build process
+    cp ${BUILD_PREFIX}/bin/protoc-23.3.0 ${PREFIX}/bin/protoc-23.3.0
+    # if [[ -x "${CONDA_PREFIX}/bin/protoc-*" ]]; then
+    #     cp ${BUILD_PREFIX}/bin/protoc-* ${PREFIX}/bin/
+    # fi
```
Comment (h-vetinari) [inline recipe/build_pytorch.sh:89]:
```suggestion
    # point to a usable protoc if we're cross-compiling
    export Protobuf_PROTOC_EXECUTABLE="$BUILD_PREFIX/bin/protoc"
```

Something like this should work in principle (it's the name of a variable that CMake should respect, but AFAICT we're calling cmake through pip)

Comment (hmaarrfk) [reply recipe/build_pytorch.sh:89]: The thing is, i've tried to specify this, in five different spellings, above...

Comment (h-vetinari) [reply recipe/build_pytorch.sh:89]: Patch it into the cmake invocation that's behind setup.py?

Comment (hmaarrfk) [reply recipe/build_pytorch.sh:89]: the 30 min iteration cycle I have makes it essentially impossible to test this for me. 

The old mac I have just isn't always plugged in.

**Comment by hmaarrfk** [comment]
Ok, so "debugging" this build on my little machine is really tiring, but I'm finding some locations in cmake that we can likely modify:


```
${PREFIX}/lib/cmake/protobuf/protobuf-targets-release.cmake
```

```
# Import target "protobuf::protoc" for configuration "Release"
set_property(TARGET protobuf::protoc APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(protobuf::protoc PROPERTIES
  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/bin/protoc-23.3.0"
  )
list(APPEND _cmake_import_check_targets protobuf::protoc )
list(APPEND _cmake_import_check_files_for_protobuf::protoc "${_IMPORT_PREFIX}/bin/protoc-23.3.0" )
```

we cna likely change the locations of the specific protoc commands.

**Comment by hmaarrfk** [comment]
seems like we got through it. fingers crossed we get to the end.

**Pull request closed without merging**